### DISCUSSION TITLE
Warm Starting ipopt

### DISCUSSION THREAD

**Discussion by tanmishra** [coin-or/Ipopt#693] [Q&A]
Hello Everyone,

I have an algorithm sketched out that should work to solve a smallish NLP to optimality if I can get a warm start to work. I tried warm starting by storing the decision/variables and again feeding back to the variable before solving the next time (kind of initialization). To be precise, I am using a for-loop for i =1 to 10. I am first solving the NLP at i=1 and then giving its solution as a warm start to i=2 and so on. But by seeing the primal infeasibility and number of iterations (both are high), I can say the warm starting is not working. It would be great if anyone could point me out correctly and help me warm start the ipopt here.
PS: I am using pyomo and ipopt as solve.

I have used these options:
        opt.options['nlp_scaling_method']= None

        results = opt.solve(model, keepfiles=keepfiles, tee=True)
        opt.options['warm_start_init_point'] = 'yes'
        opt.options['warm_start_bound_frac'] = 1e-6
        opt.options['warm_start_bound_push'] = 1e-6
        opt.options['warm_start_mult_bound_push'] = 1e-6
        opt.options['warm_start_slack_bound_frac'] = 1e-6
        opt.options['warm_start_slack_bound_push'] = 1e-6

**Comment by svigerske** [comment]
I don't know much about the Pyomo/Ipopt link. You could increase the print_level of Ipopt to debug more what is happening. At a sufficiently high level, it would also print the initial point that it received. If it isn't as expected, then you may have to check with the Pyomo developers.

If Pyomo relies on Ipopt's AMPL link, then Pyomo would need to ensure that some suffixes are stored in the `.nl` file that hold your values for the bound multipliers, see https://coin-or.github.io/Ipopt/SPECIALS.html#AMPL_WARMSTART
